bin/reth/src/args/types.rs

```diff
@@ -2,19 +2,28 @@
 
 use std::{fmt, str::FromStr};
 
-/// A helper type that maps `0` to `None` when parsing CLI arguments.
+/// A helper type that maps `u32::MAX` to `None` when parsing CLI arguments.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct ZeroAsNone(pub Option<u64>);
+pub struct MaxAsNone(pub Option<u32>);
```

Comment:
what we want here is something like 

`MaxU32(u32)`

where fromStr does "max" => u32::MAX

Reply:
> what we want here is something like
> 
> `MaxU32(u32)`
> 
> where fromStr does "max" => u32::MAX

ty for this, does it look better?